lib/utils.ex

```diff
@@ -8,17 +8,21 @@ defmodule HashRing.Utils do
   end
   def hash(key), do: hash("#{key}")
 
+  @spec gen_items([{binary, integer}]) :: [{integer, binary}]
   @spec gen_items(binary, integer) :: [{integer, binary}]
+  def gen_items([]), do: []
+  def gen_items(nodes), do: do_gen_items(nodes, [])
   def gen_items([], _num_replicas), do: []
-  def gen_items(nodes, num_replicas) do
-    gen_items(nodes, Enum.to_list(0..(num_replicas - 1)), [])
+  def gen_items(nodes, default_num_replicas) do
+    nodes = for node <- nodes, do: {node, default_num_replicas}
+    gen_items(nodes)
```

Comment:
call do_gen_items here?

Reply:
eh, the thought is this function transforms the input into a value acceptable for `gen_items/1`